File: 392-is-subsequence/392.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-


class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        len1, len2 = len(s), len(t)
        i, j = 0, 0
        while (i < len1 and j < len2):
            if s[i] == t[j]:
                i += 1
            j += 1
        return i == len1


if __name__ == '__main__':
    sol = Solution()
    print(sol.isSubsequence("abc", "ahbgdc"))
    print(sol.isSubsequence("axc", "ahbgdc"))

File: shoope/392-is-subsequence/392.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        int len1 = s.length(), len2 = t.length();
        int i = 0, j = 0;
        while (i < len1 && j < len2) {
            if (s[i] == t[j]) {
                ++i;
            }
            ++j;
        }
        return i == len1;
    }
};

int main(int argc, char** args) {
    Solution sol = Solution();
    cout << sol.isSubsequence("abc", "ahbgdc") << endl;
    cout << sol.isSubsequence("axc", "ahbgdc") << endl;
    return 0;
}